common/src/app/common/spec/shape.cljc

```diff
@@ -202,12 +202,52 @@
 (s/def :internal.shape.text/content
   (s/nilable
    (s/or :text-container
-         (s/keys :req-un [:internal.shape.text/type
-                          :internal.shape.text/children]
-                 :opt-un [:internal.shape.text/key])
+         (s/keys :req-un [:internal.shape.text/type]
+                 :opt-un [:internal.shape.text/key
+                          :internal.shape.text/children])
          :text-content
          (s/keys :req-un [:internal.shape.text/text]))))
 
+(s/def :internal.shape.text/position-data
+  (s/coll-of :internal.shape.text/position-data-element
+             :kind vector?
+             :min-count 1))
+
+(s/def :internal.shape.text/position-data-element
+  (s/keys :req-un [:internal.shape.text.position-data/x
+                   :internal.shape.text.position-data/y
+                   :internal.shape.text.position-data/width
+                   :internal.shape.text.position-data/height]
+          :opt-un [:internal.shape.text.position-data/fill-color
+                   :internal.shape.text.position-data/fill-opacity
+                   :internal.shape.text.position-data/font-family
+                   :internal.shape.text.position-data/font-size
+                   :internal.shape.text.position-data/font-style
+                   :internal.shape.text.position-data/font-weight
+                   :internal.shape.text.position-data/rtl?
+                   :internal.shape.text.position-data/text
+                   :internal.shape.text.position-data/text-decoration
+                   :internal.shape.text.position-data/text-transform]
+          ))
+
+(s/def :internal.shape.text.position-data/x ::us/safe-number)
+(s/def :internal.shape.text.position-data/y ::us/safe-number)
+(s/def :internal.shape.text.position-data/width ::us/safe-number)
+(s/def :internal.shape.text.position-data/height ::us/safe-number)
+
+(s/def :internal.shape.text.position-data/fill-color ::fill-color)
+(s/def :internal.shape.text.position-data/fill-opacity ::fill-opacity)
+(s/def :internal.shape.text.position-data/fill-color-gradient ::fill-color-gradient)
+
+(s/def :internal.shape.text.position-data/font-family string?)
+(s/def :internal.shape.text.position-data/font-size string?)
+(s/def :internal.shape.text.position-data/font-style string?)
+(s/def :internal.shape.text.position-data/font-weight string?)
+(s/def :internal.shape.text.position-data/rtl? boolean?)
```

Comment:
I think we have decided not use the `?` ending on props on the file data structures. (I know that we already have some of them hehe)

Reply:
OK. I'll change it